File: php/tests/HostIntegrationTest.php
<?php

declare(strict_types=1);

namespace NotifyService\Tests;

use NotifyService\Api\Model\NotificationChannel;
use NotifyService\Api\Model\Recipient;
use NotifyService\Api\Model\SendNotificationRequest;
use NotifyService\NotifyClient;
use PHPUnit\Framework\TestCase;

final class HostIntegrationTest extends TestCase
{
    private string $baseUrl;
    private string $apiKey;

    protected function setUp(): void
    {
        $baseUrl = getenv('NOTIFY_BASE_URL');
        $apiKey  = getenv('NOTIFY_API_KEY');
        if ($baseUrl === false || $apiKey === false || $baseUrl === '' || $apiKey === '') {
            self::markTestSkipped('NOTIFY_BASE_URL/NOTIFY_API_KEY not set.');
        }
        $this->baseUrl = $baseUrl;
        $this->apiKey  = $apiKey;
    }

    public function testSendThenList(): void
    {
        $notify = new NotifyClient($this->baseUrl, $this->apiKey);

        $recipient = (new Recipient())
            ->setAddress('alice@example.com')
            ->setName('Alice');

        $request = (new SendNotificationRequest())
            ->setChannel(NotificationChannel::EMAIL)
            ->setRecipient($recipient)
            ->setSubject('Integration')
            ->setHtmlBody('<p>Integration test</p>');

        $created = $notify->notifications->sendNotification($request);
        $page    = $notify->notifications->listNotifications(null, null, null, null, 5);
        $ids     = array_map(static fn ($n) => $n->getId(), $page->getData());

        self::assertContains($created->getId(), $ids);
    }
}

File: java/src/test/java/com/notifyservice/HostIntegrationTest.java
package com.notifyservice;

import static org.junit.jupiter.api.Assertions.assertTrue;

import com.notifyservice.api.model.NotificationChannel;
import com.notifyservice.api.model.Recipient;
import com.notifyservice.api.model.SendNotificationRequest;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;

@EnabledIfEnvironmentVariable(named = "NOTIFY_BASE_URL", matches = ".+")
@EnabledIfEnvironmentVariable(named = "NOTIFY_API_KEY", matches = ".+")
class HostIntegrationTest {

    private static NotifyClient client() {
        return new NotifyClient(System.getenv("NOTIFY_BASE_URL"), System.getenv("NOTIFY_API_KEY"));
    }

    @Test
    void sendThenList() throws Exception {
        NotifyClient notify = client();
        SendNotificationRequest req = new SendNotificationRequest()
                .channel(NotificationChannel.EMAIL)
                .recipient(new Recipient().address("alice@example.com").name("Alice"))
                .subject("Integration")
                .htmlBody("<p>via java SDK</p>");

        var created = notify.notifications.sendNotification(req, null);
        var page = notify.notifications.listNotifications(null, null, null, null, 5, null);

        assertTrue(page.getData().stream().anyMatch(n -> n.getId().equals(created.getId())));
    }
}
